### PR TITLE
Search box and versions.json

### PR DESCRIPTION
**Pull request by johanandren**
Fixes searching (and makes versions.json automagic) #389

### CODE REVIEW

File: resources/javascript/docs-search.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+# front matter to make jekyll process it
+layout: null
+---
+
+$(function () {
```
Comment (johanandren) [inline resources/javascript/docs-search.js]: Aaah jquery, the most polymorphic method in the history of mankind. Passing it a function like this means: run function when page has loaded, if someone wonders.

File: resources/javascript/akka.io.js [hunk 2]
```diff
@@ -1,106 +0,0 @@
-var urlArr = window.location.href.split("/");
```
Comment (johanandren) [inline resources/javascript/akka.io.js]: This and below was mostly the old logic for populating the downloads page with dependency-template to copy paste into sbt/maven/gradle.

File: _includes/footer.html [hunk 3]
```diff
@@ -4,14 +4,13 @@
...
-  <script src="{{ site.baseurl }}/resources/javascript/mini-toggler.js" type="text/javascript"></script>
-  <script src="{{ site.baseurl }}/resources/javascript/stickySearch.js" type="text/javascript"></script>
-  <script src="{{ site.baseurl }}/resources/javascript/tabPanel.js" type="text/javascript"></script>
   <!-- highlight js -->
   <link href="{{ site.baseurl }}/resources/stylesheets/styles/github.css" type="text/css" rel="stylesheet" />
   <script src="{{ site.baseurl }}/resources/javascript/highlight.pack.js" type="text/javascript"></script>
+  <!-- algolia search -->
+  <script src="https://cdn.jsdelivr.net/docsearch.js/1/docsearch.min.js" type="text/javascript"></script>
+  <script src="{{ site.baseurl }}/resources/javascript/docs-search.js" type="text/javascript"></script>
+
```
Comment (johanandren) [inline _includes/footer.html]: Would like to only load it on the docs page where it is used, but the it depends on jQuery which is included in the footer include and which then contains the end html tag, and didn't make a bigger change for that.

File: _config.yml [hunk 4]
```diff
@@ -22,7 +22,8 @@ defaults:
     path: "" # all files
   values:
     # note that changing this requires a jekyll restart to be picked up if serving locally
-    # it does not generate a new versions.json so that you will have to update manually if changing here
+    # versions.json will be updated as long as the changes here are patch versions, for minor and major it needs to
+    # be updated manually
```
Comment (ktoso) [inline _config.yml]: very cool

File: _includes/footer.html (hunk 3 above)
Comment (ktoso) [inline _includes/footer.html]: I wonder if it matters much - how big is it?

File: resources/javascript/docs-search.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
+---
+# front matter to make jekyll process the file
+layout: null
+---
+
+$(function () {
+
+    // set by jekyll
+    var latestAkkaVersion = "{{ page.current_akka_version }}";
```
Comment (ktoso) [inline resources/javascript/docs-search.js]: <3

File: versions.json [hunk 6]
```diff
@@ -36,17 +40,19 @@
...
     },
     "2.4": {
-      "latest": "2.4.17",
+      "latest": "2.4.18",
       "outdated": true,
       "instead": "2.5."
     },
     "2.5-": {
-      "latest": "2.5.0",
+      // updated by jekkyll
```
Comment (ktoso) [inline versions.json]: comments in .json? isn't that invalid? WIll jekyll remove those?

Comment (johanandren) [inline versions.json]: ah, good catch, that is indeed invalid

File: _includes/footer.html (hunk 3 above)
Comment (johanandren) [inline _includes/footer.html]: The algolia one is 144kb, our script is ~1kb, with cleared caches it loads in 13ms and parses/evaluates in ~2.5ms. Was more about cleanliness and not loading things you do not use than performance related.

File: versions.json [hunk 7]
```diff
@@ -40,7 +40,7 @@ layout: null
       "instead": "2.5."
     },
     "2.4": {
-      "latest": "2.4.18",
+      "latest": "{{ page.previous_akka_version }}",
```
Comment (ktoso) [inline versions.json]: 👍

Comment (ktoso) [inline versions.json]: "you're a wizard, harry!"

**Pull request merged**